### PR TITLE
[Frontend] Integrate resources api

### LINKED ISSUE #6: History Management
# As a user who wants to monitor the progress, I want to access a record of my past projects and the specific hardware configurations I utilized for each one.

### PR DESCRIPTION
**Pull request by weii41392**
## Changes made
- Since backend APIs are delayed, I wrote a [fake server](https://github.com/collebrusco/ECE-461L-Orange/tree/integrate-resources-api/fake-server) to test with the frontend. The server has in-memory storage and doesn't connect with the database. For the simplicity, it doesn't comply with the project requirement and is merely for testing purposes.
- Integrate `resources` APIs into frontend

## How to test
- Run the following scripts to build & run containers. Since this is for testing purpose, I leave docker compose unchanged.
```
cd fake-server/
. ./run.sh
cd ../frontend/
. ./build.sh
```
- View the result in [http://127.0.0.1:3000/](http://127.0.0.1:3000/). Due to the constraints of some browsers, `localhost` won't work in some cases. It is safer to use `127.0.0.1`.

### CODE REVIEW

**Review by collebrusco — COMMENTED** [review]
I tried to test this, but am unable to login. I get a 'signed in successfully' message but get stuck on the login screen

**Comment by yawn03** [comment]
Looks good!

File: fake-server/views/constants.py
Comment (yawn03) [inline fake-server/views/constants.py:1]: I really like how you separated out responses into constants!

File: fake-server/views/projects.py
Comment (yawn03) [inline fake-server/views/projects.py:1]: You technically don't need to call jsonify. If you set the response type to application/json you can just return the dict and it'll automatically handle the conversion.

**Review by yawn03 — APPROVED** [review]

**Pull request merged**